data/importexport/document_downloader.py

```diff
@@ -0,0 +1,66 @@
+import os
+from time import sleep
+from qgis.core import QgsNetworkAccessManager
+from PyQt4.QtNetwork import (
+    QNetworkAccessManager,
+    QNetworkRequest
+)
+from PyQt4.QtGui import (
+    QApplication,
+)
+from PyQt4.QtCore import (
+    QUrl,
+    QByteArray,
+    QEventLoop,
+)
+
+class Doc_Downloader(object):
+    """
+    this class 
+    """
+    def __init__(self, username, password):
+        #self.username = username
+        #self.password = password
+        self.manager = QNetworkAccessManager()
+        self.manager.finished.connect(self.handle_download)
+        self.header_data = QByteArray(
+            '{}:{}'.format(username, password)
+        ).toBase64()
+
+    def kobo_download(self, src_url, dest_filename):
+        self.download_result = None
+        self.dest_filename = dest_filename
+        request = QNetworkRequest(QUrl(src_url))
+        request.setRawHeader(
+            'Authorization',
+            'Basic {}'.format(self.header_data)
+        )
+        try:
+            self.manager.get(request)
+            self.loop = QEventLoop()
+            self.loop.exec_()
+        except:
+            pass
+        
+        self.loop = None
+        sleep(1)
+        QApplication.processEvents()
+        if self.download_result is None:
+            return False
+        return self.download_result
+            
+    def handle_download(self, reply):
+        if reply.errorString() != 'Unknown error':
+            self.download_result = False
+        else:
```

Comment:
Does this mean that this code block will be executed when the error string is "Unknown error"?

Reply:
Dear @pgathogo ,
Kindly be informed that after trying many times found that this component initialize this attribute to save error messages while downloading files and if the file is downloaded successfully then reply.errorString() should return 'Unknown error' which is the default value

Reply:
Dear @pgathogo  ,
Thank your very much! Truly appreciate your support.